### PR TITLE
Add ASM search box from AOS

### PR DESCRIPTION
**Pull request by adrianbj**

### CODE REVIEW

**Comment by adrianbj** [comment]
Just wanted to note with this one that there are a couple of sub options for this in AOS:
image

I don't personally use them, so they aren't enabled, but I am not sure how RockAdminTweaks should attempt to handle config options for tweaks moving forward because I can see it being an issue for some other tweaks as well. Any ideas?

**Comment by adrianbj** [comment]
@BernhardBaumrock - Perhaps you could add a new "tweak settings" parameter that can be populated with an array of settings. These would be saved to the module's config and be checked with the tweak's code logic. Perhaps the keys from this settings parameter would be prefixed by the class name of the tweak to avoid possible conflicts.

**Comment by BernhardBaumrock** [comment]
Hi @adrianbj thx again! I knew this would come up one day... I'm not really keen on implementing such a feature, as I think it's then much more going into the direction of a regular module rather than a tweak. But I thought we'd maybe allow custom settings via $config->myTweakSettingWhatever = true;
What do you think?

**Comment by adrianbj** [comment]
@BernhardBaumrock - the problem of course with the $config approach for settings is figuring out how to let users know what settings are available for each tweak. Perhaps another parameter where these could be listed and appear under each tweak?

**Comment by BernhardBaumrock** [comment]
That's not a problem IMHO once we have another "help" param for the info() array that supports markdown. Then any instructions can be put there and every user get's instant help and description of the features and settings of the tweak.

File: tweaks/Inputfields/AsmSearchBox/select2/js/select2-v4.0.2.patch
Comment (BernhardBaumrock) [inline tweaks/Inputfields/AsmSearchBox/select2/js/select2-v4.0.2.patch:1]: Hey @adrianbj what is this patch file doing? Is it necessary?

File: tweaks/Inputfields/AsmSearchBox/select2/js/select2.zip
Comment (BernhardBaumrock) [inline tweaks/Inputfields/AsmSearchBox/select2/js/select2.zip:1]: @adrianbj  Do we need the ZIP as part of the tweak?

File: tweaks/Inputfields/AsmSearchBox/select2/js/select2-v4.0.2.patch
Comment (adrianbj) [reply tweaks/Inputfields/AsmSearchBox/select2/js/select2-v4.0.2.patch:1]: @tpr mentioned that he had to modify select2 for it to work for this tweak so it looks like a good way to keep a record of the changes he had to make.

File: tweaks/Inputfields/AsmSearchBox/select2/js/select2.zip
Comment (adrianbj) [reply tweaks/Inputfields/AsmSearchBox/select2/js/select2.zip:1]: Happy to have this deleted - I just copied across what @tpr had for the assets for the tweak in AOS.

**Pull request merged**